promscale/send-data/remote-write.md

```diff
@@ -0,0 +1,302 @@
+payload is smaller. We recommend this format if you're building a system to
+ingest a lot of data continuously.
+
+<procedure>
+
+### Using the Promscale endpoint with the Protobuf format
+
+1.  Fetch the protocol buffer definition files [from the Prometheus GitHub
+    repo][protobuf-definition].
+1.  Compile the definitions into the structures of the programming language
+    you're using. For instructions, see the [Google Protobuf
+    tutorials][protobuf-tutorials].
+1.  Use the structures to construct requests to send to the Promscale write
+    endpoint. For more information, see [a sample request written in
+    Go][go-sample].
+
+</procedure>
+
+### Protobuf example in Go
+
+Before you begin, you need the Protobuf definition files, available in the
+Prometheus codebase. These definitions must be compiled into the necessary
+classes for payload serialization.
+
+Since Prometheus is written in the Go language, you can directly import the
+pre-generated classes from the [prompb] package in the Prometheus repository.
+This gives you all the necessary classes to create a write request.
+
+Then, you can:
+
+*   Create a `WriteRequest` object
+*   Preload it with metrics data
+*   Serialize it
+*   Send it as an HTTP request
+
+A simplified `main.go` file for these steps looks like this:
+
+```go
+package main
+import (
+    "bytes"
+    "net/http"
+    "github.com/gogo/protobuf/proto"
+    "github.com/golang/snappy"
+)
+func main() {
+    // Create the WriteRequest object with metric data populated.
+    wr := &WriteRequest{
+        Timeseries: []TimeSeries{
+            TimeSeries{
+                Labels: []Label{
+                    {
+                        Name:  "__name__",
+                        Value: "foo",
+                    },
+                },
+                Samples: []Sample{
+                    {
+                        Timestamp: 1577836800000,
+                        Value:     100.0,
+                    },
+                },
+            },
+        },
+    }
+    // Marshal the data into a byte slice using the protobuf library.
+    data, err := proto.Marshal(wr)
+    if err != nil {
+        panic(err)
+    }
+    // Encode the content into snappy encoding.
+    encoded := snappy.Encode(nil, data)
+    body := bytes.NewReader(encoded)
+    // Create an HTTP request from the body content and set necessary parameters.
+    req, err := http.NewRequest("POST", "http://localhost:9201/write", body)
+    if err != nil {
+        panic(err)
+    }
+    // Set the required HTTP header content.
+    req.Header.Set("Content-Type", "application/x-protobuf")
+    req.Header.Set("Content-Encoding", "snappy")
+    req.Header.Set("X-Prometheus-Remote-Write-Version", "0.1.0")
+    // Send request to Promscale.
+    resp, err := http.DefaultClient.Do(req)
+    if err != nil {
+        panic(err)
+    }
+    defer resp.Body.Close()
+    ...
+```
+
+## JSON streaming format
+
```

Comment:
Not sure if it's GH's diff, or if there's a non-printing char or something breaking this.